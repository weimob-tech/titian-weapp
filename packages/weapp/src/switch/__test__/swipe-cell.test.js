const simulate = require('miniprogram-simulate');
const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('switch');
    comp.triggerLifeTime('attached');

    expect(comp).toMatchSnapshot();
  });

  it('should render success props', async () => {
    const comp = getComponent('switch', {
      value: true
    });
    comp.triggerLifeTime('attached');

    expect(comp).toMatchSnapshot();
    comp.setData({ value: false });
    await simulate.sleep(0);
    expect(comp).toMatchSnapshot();
  });
});
