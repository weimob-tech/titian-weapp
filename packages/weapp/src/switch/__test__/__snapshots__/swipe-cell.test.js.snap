// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <wx-view
    $gdc="{{[Function]}}"
    $wxs:bind:tap="{{[Function]}}"
    $wxs:bind:touchend="{{[Function]}}"
    $wxs:bind:touchmove="{{[Function]}}"
    $wxs:bind:touchstart="{{[Function]}}"
    $wxs:change:value="{{[Function]}}"
    class="titian-switch  ext-class"
    data-active-color=""
    data-color=""
    data-disabled="{{false}}"
    data-loading="{{false}}"
    data-value="{{null}}"
    style=";width:70rpx;height:40rpx;padding:0 6rpx;border-radius:40rpx"
    value="{{true}}"
  >
    <wx-view
      class="titian-switch-ball"
      style="width:28rpx;height:28rpx;border-radius:28rpx"
    />
  </wx-view>
</main>
`;

exports[`test component should render success props 1`] = `
<main>
  <wx-view
    $gdc="{{[Function]}}"
    $wxs:bind:tap="{{[Function]}}"
    $wxs:bind:touchend="{{[Function]}}"
    $wxs:bind:touchmove="{{[Function]}}"
    $wxs:bind:touchstart="{{[Function]}}"
    $wxs:change:value="{{[Function]}}"
    class="titian-switch  ext-class"
    data-active-color=""
    data-color=""
    data-disabled="{{false}}"
    data-loading="{{false}}"
    data-value="{{true}}"
    style=";width:70rpx;height:40rpx;padding:0 6rpx;border-radius:40rpx"
    value="{{true}}"
  >
    <wx-view
      class="titian-switch-ball"
      style="width:28rpx;height:28rpx;border-radius:28rpx"
    />
  </wx-view>
</main>
`;

exports[`test component should render success props 2`] = `
<main>
  <wx-view
    $gdc="{{[Function]}}"
    $wxs:bind:tap="{{[Function]}}"
    $wxs:bind:touchend="{{[Function]}}"
    $wxs:bind:touchmove="{{[Function]}}"
    $wxs:bind:touchstart="{{[Function]}}"
    $wxs:change:value="{{[Function]}}"
    class="titian-switch  ext-class"
    data-active-color=""
    data-color=""
    data-disabled="{{false}}"
    data-loading="{{false}}"
    data-value="{{false}}"
    style=";width:70rpx;height:40rpx;padding:0 6rpx;border-radius:40rpx"
    value="{{false}}"
  >
    <wx-view
      class="titian-switch-ball"
      style="width:28rpx;height:28rpx;border-radius:28rpx"
    />
  </wx-view>
</main>
`;
